xdsl/utils/lexer.py

```diff
@@ -7,6 +7,8 @@
 from string import hexdigits
 from typing import Literal, TypeAlias, TypeGuard, cast, overload
 
+import numpy as np
```

Comment:
same here, numpy is not a core dependency, we need to use other helpers for functionality below